vision/src/autogluon/vision/detector/detector.py

```diff
@@ -0,0 +1,379 @@
+        task._logger.setLevel(log_level)
+        task._logger.propagate = True
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            self._detector = task.fit(train_data, tuning_data, 1 - holdout_frac, random_state)
+        self._detector._logger.setLevel(log_level)
+        self._detector._logger.propagate = True
+        self._fit_summary = task.fit_summary()
+        return self
+
+    def _validate_kwargs(self, kwargs):
+        """validate and initialize default kwargs"""
+        kwargs['holdout_frac'] = kwargs.get('holdout_frac', 0.1)
+        if not (0 < kwargs['holdout_frac'] < 1.0):
+            raise ValueError(f'Range error for `holdout_frac`, expected to be within range (0, 1), given {kwargs["holdout_frac"]}')
+        kwargs['random_state'] = kwargs.get('random_state', None)
+        kwargs['nthreads_per_trial'] = kwargs.get('nthreads_per_trial', None)
+        kwargs['ngpus_per_trial'] = kwargs.get('ngpus_per_trial', None)
+        if kwargs['ngpus_per_trial'] is not None and kwargs['ngpus_per_trial'] > 0:
+            detected_gpu = get_gpu_count()
+            if detected_gpu < kwargs['ngpus_per_trial']:
+                raise ValueError(f"Insufficient detected # gpus {detected_gpu} vs requested {kwargs['ngpus_per_trial']}")
+        # tune kwargs
+        hpo_tune_args = kwargs.get('hyperparameter_tune_kwargs', {})
+        hpo_tune_args['num_trials'] = hpo_tune_args.get('num_trials', 1)
+        hpo_tune_args['search_strategy'] = hpo_tune_args.get('search_strategy', 'random')
+        if not hpo_tune_args['search_strategy'] in ('random', 'bayesopt', 'grid'):
+            raise ValueError(f"Invalid search strategy: {hpo_tune_args['search_strategy']}, supported: ('random', 'bayesopt', 'grid')")
+        hpo_tune_args['max_reward'] = hpo_tune_args.get('max_reward', None)
+        if hpo_tune_args['max_reward'] is not None and hpo_tune_args['max_reward'] < 0:
+            raise ValueError(f"Expected `max_reward` to be a positive float number between 0 and 1.0, given {hpo_tune_args['max_reward']}")
+        hpo_tune_args['scheduler_options'] = hpo_tune_args.get('scheduler_options', None)
+        kwargs['hyperparameter_tune_kwargs'] = hpo_tune_args
+        return kwargs
+
+    def predict(self, data, as_pandas=True):
+        """Predict objects in image, return the confidences, bounding boxes of each predicted object.
+
+        Parameters
+        ----------
+        data : str, pd.DataFrame or ndarray
+            The input data, can be str(filepath), pd.DataFrame with 'image' column, or raw ndarray input.
+        as_pandas : bool, default = True
+            Whether to return the output as a pandas object (True) or list of numpy array(s) (False).
+            Pandas object is a DataFrame.
+
+        Returns
+        -------
+
+        pd.DataFrame
+            The returned dataframe will contain (`pred_score`, `pred_bbox`, `pred_id`).
```

Comment:
Should `pred_id` just be the index? Does it need to be its own column?

Reply:
this is the id of each predicted object